tests/testthat/test-regression100Kcases.R

```diff
@@ -0,0 +1,125 @@
+set.seed(111)
+p.noise <- 15
+noise <- as.data.frame(replicate(p.noise, rnorm(nrow(vehicles))))
+colnames(noise) <- paste0("noise", 1:p.noise)
+vehicles <- cbind(vehicles, noise)
+
+test_that("Linear regression",
+{
+    expect_is(Regression(log.price~.-region_weight, data = vehicles),
+              "LinearRegression")
+    ## Effects Plot doesn't like "." in formula
+    expect_error(print(Regression(log.price ~ drive+condition+log.odometer+region+year,
+                   data = vehicles,
+                   weights = region_weight,
+                   output = "Effects Plot")), NA)
+    expect_error(print(Regression(log.price~.-region_weight,
+                   data = vehicles,
+                   weights = region_weight,
+                   output = "ANOVA")), NA)
+
+    expect_is(Regression(log.price~.-region_weight, data = vehicles,
+                         weights = region_weight),
+              "LinearRegression")
+})
+
+test_that("Weighted logistic regression", {
+    vehicles$great.condition <- factor(as.integer(vehicles$condition == "excellent" |
+                                           vehicles$condition == "like new"))
+    expect_is(Regression(great.condition~.-region_weight-condition,
+                         weight = region_weight,
+                         data = vehicles, type = "Binary Logit"),
+              "BinaryLogitRegression")
+    expect_is(Regression(great.condition~drive+region+year,
+                         weight = region_weight, output = "Jaccard Coefficient",
+                         data = vehicles, type = "Linear"),
+              "LinearRegression")
+
+    vehicles$great.condition <- NULL
+})
+
+test_that("Weighted regression with count response", {
+    vehicles$count <- rpois(n = nrow(vehicles), lambda = exp(as.numeric(vehicles$condition)))
+    expect_is(Regression(count~.-region_weight,
+                         weight = region_weight,
+                         data = vehicles, type = "Poisson"),
+              "PoissonRegression")
+    expect_is(Regression(count~.-region_weight,
+                         weight = region_weight,
+                         data = vehicles, type = "Quasi-Poisson"),
+              "QuasiPoissonRegression")
+    vehicles$count <- MASS::rnegbin(n = nrow(vehicles),
+                                    mu = exp(as.numeric(vehicles$condition)),
+                                    theta = 2)
+    expect_is(Regression(count~.-region_weight,
+                         data = vehicles, type = "NBD"),
+              "NBDRegression")  ## unweighted
+    expect_is(Regression(count~.-region_weight,
+                         weight = region_weight,
+                         data = vehicles, type = "NBD"),
+              "NBDRegression")
+
+    vehicles$count <- NULL
+})
+
+test_that("Multinomial response",
+{
+    expect_is(Regression(year~.-region_weight,
+                         data = vehicles, type = "Multinomial Logit"),
+              "MultinomialLogitRegression")
+
```

Comment:
weighted multinomial logit is missing in the tests.

Reply:
Added